### ISSUE TITLE
Issues when combining webpack HMR and react-refresh-webpack-plugin

### ISSUE THREAD

**Issue by JKHSDTV** [pmmmwh/react-refresh-webpack-plugin#807]
We have both HMR and RRWP running, and both seem to work fine on their own. But in a project with both, we are facing the problem that `.js`-files can't be reloaded without a full reload. I suspect this may be due to the chaining, so before I dig deeper, is the following supposed to work in the first place ? Or are we 'doing things wrong' somehow ?

- we have a `main.js`, which after some initializing calls `ReactDOM.render(<Provider store = {store} ><ClientContent /></Provider>, document.getElementById('react_container'));`.
- in our component chain, we have many components/containers in the form of a `Foo.jsx` component wrapped by a `FooContainer.js` container
- during initializing of props for `Foo`'s, the `FooContainer.js` `import`'s multiple functions from `WhatEverStuff.js`
- changes to `WhatEverStuff.js` can not be hot reloaded

Since our setup works without RRWP, I suspect that it gets confused somewhere in that chain ? Any hints ?

**Comment by nyngwang** [comment]
What is RRWP?

**Comment by JKHSDTV** [comment]
Yes, someone on IRC referenced it like that and I was under the impression that it was a common abbreviation.

**Comment by nyngwang** [comment]
I will need to see your webpack.config.js. The setup itself is tricky.

**Comment by JKHSDTV** [comment]
```json
[...]
devtool: 'eval-source-map',
mode: 'development',
devServer: {
  hot: true,
  liveReload: false,
  allowedHosts: [ 'all' ],
  devMiddleware: { writeToDisk: true, index: false },
  proxy: [{
    context: [
      '**',
      '!/client/our-client.js'
    ],
    target: 'http://localhost:8000',
    proxyTimeout: 1000 * 60 * 30,
    timeout: 1000 * 60 * 30
  }],
  compress: false,
  port: 9000
}
```

And we run `webpack` like so: `webpack serve --config=webpack.development.config.js`

**Comment by nyngwang** [comment]
@JKHSDTV Does your `<ClientContent />` directly contain those many `Foo.jsx` component wrapped by a `FooContainer.js` container? If so try adding one more layer like this:

```
<App/> -> <ClientContent/> -> many `Foo.jsx`
```

**Comment by JKHSDTV** [comment]
Got multiple levels like that between Foo and ClientContent, and no anonymous default exports. Went through all the troubleshooting, none seem to apply.

**Comment by pmmmwh** [comment]
Are you doing any manual `module.hot.accept`?

**Comment by pmmmwh** [comment]
It would be helpful if a reproduction can be provided

**Comment by JKHSDTV** [comment]
> Are you doing any manual `module.hot.accept`?

Your responded to a comment from me about how I use `module.hot.accept` 😂 

I'm afraid it's the typical situation of where it's part of a complex proprietary setup that isn't easy to export as a test case :(

**Comment by pmmmwh** [comment]
The point of this plugin is to not have to do `module.hot.accept` - it will accept (or not) based on the rules of React Refresh. I will consider this "working as expected" and won't fix.

**Comment by Gaibhne** [comment]
I'm not sure I understand your reply, @pmmmwh - are you saying that when using React Refresh, hot reloading regular `.js` files is not possible at all, or that React Refresh should be doing the reloading of regular `.js` files ?

**Comment by pmmmwh** [comment]
> I see, I was not aware of that, thank you! Any hint on where I could read up on how to configure that behaviour with React Refresh ? For us, all changes to stuff `import`ed from `.js` files trigger a full app reload, even if it's just the value of a constant declared there.

I cannot give concrete advice without looking at the code, but the one way of potentially making it work better is to ensure these don't propagate up to the app root by introducing component boundaries. For example, if your root file always import a constant, that will always trigger a reload since we do not know from looking at the value that it is safe to not bail out, and it is already at the root level.
